test/e2e/apimachinery/discovery.go

```diff
@@ -155,4 +157,118 @@ var _ = SIGDescribe("Discovery", func() {
+			apiResource   string
+			apiVersion    string
+			groupVersion  string
+			validResource string
+		}{
+			{
+				apiBasePath:   "/apis/",
+				apiResource:   "apps",
+				apiVersion:    "v1",
+				groupVersion:  "apps/v1",
+				validResource: "deployments",
+			},
+			{
+				apiBasePath:   "/apis/",
+				apiResource:   "autoscaling",
+				apiVersion:    "v1",
+				groupVersion:  "autoscaling/v1",
+				validResource: "horizontalpodautoscalers",
+			},
+			{
+				apiBasePath:   "/apis/",
+				apiResource:   "autoscaling",
+				apiVersion:    "v2",
+				groupVersion:  "autoscaling/v2",
+				validResource: "horizontalpodautoscalers",
+			},
+			{
+				apiBasePath:   "/apis/",
+				apiResource:   "apiregistration.k8s.io",
+				apiVersion:    "v1",
+				groupVersion:  "apiregistration.k8s.io/v1",
+				validResource: "apiservices",
+			},
+			{
+				apiBasePath:   "/apis/",
+				apiResource:   "authentication.k8s.io",
+				apiVersion:    "v1",
+				groupVersion:  "authentication.k8s.io/v1",
+				validResource: "tokenreviews",
+			},
+			{
+				apiBasePath:   "/apis/",
+				apiResource:   "authorization.k8s.io",
+				apiVersion:    "v1",
+				groupVersion:  "authorization.k8s.io/v1",
+				validResource: "selfsubjectaccessreviews",
+			},
+			{
+				apiBasePath:   "/apis/",
+				apiResource:   "batch",
+				apiVersion:    "v1",
+				groupVersion:  "batch/v1",
+				validResource: "jobs",
+			},
+			{
+				apiBasePath:   "/apis/",
+				apiResource:   "coordination.k8s.io",
+				apiVersion:    "v1",
+				groupVersion:  "coordination.k8s.io/v1",
+				validResource: "leases",
+			},
+			{
+				apiBasePath:   "/api",
+				apiResource:   "",
+				apiVersion:    "v1",
+				groupVersion:  "v1",
+				validResource: "namespaces",
+			},
+			{
+				apiBasePath:   "/apis/",
+				apiResource:   "events.k8s.io",
+				apiVersion:    "v1",
+				groupVersion:  "events.k8s.io/v1",
+				validResource: "events",
+			},
+			{
+				apiBasePath:   "/apis/",
+				apiResource:   "policy",
+				apiVersion:    "v1",
+				groupVersion:  "policy/v1",
+				validResource: "poddisruptionbudgets",
+			},
+			{
+				apiBasePath:   "/apis/",
+				apiResource:   "scheduling.k8s.io",
+				apiVersion:    "v1",
+				groupVersion:  "scheduling.k8s.io/v1",
+				validResource: "priorityclasses",
+			},
+		}
+
+		for _, t := range tests {
+			resourceList := &metav1.APIResourceList{}
+			apiPath := t.apiBasePath + t.apiResource + "/" + t.apiVersion
+			ginkgo.By(fmt.Sprintf("Requesting APIResourceList from %q", apiPath))
+			err := f.ClientSet.Discovery().RESTClient().Get().AbsPath(apiPath).Do(ctx).Into(resourceList)
+			framework.ExpectNoError(err, "Fail to access: %s", apiPath)
+			gomega.Expect(resourceList.GroupVersion).To(gomega.Equal(t.groupVersion))
+
+			foundResource := false
+			for _, r := range resourceList.APIResources {
+				if t.validResource == r.Name {
+					foundResource = true
+					break
+				}
+			}
+			gomega.Expect(foundResource).To(gomega.BeTrue())
```

Comment:
if this fails, the content of `resourceList.APIResources` would be helpful to debug

Reply:
```
gomega.Expect(foundResource).To(gomega.BeTrue(), "Resource %q was not found inside of resourceList\n%#v", t.validResource, resourceList.APIResources)
```
If this now fails it will produce something like the following in the logs 
```
  [FAILED] Resource "..." was not found inside of resourceList
  []v1.APIResource{v1.APIResource{...}}
  Expected
      <bool>: false
  to be true
  In [It] at: /home/heyste/go/src/k8s.io/kubernetes/test/e2e/apimachinery/discovery.go:259
```